docassemble/ALToolbox/al_income.py

```diff
@@ -708,33 +805,39 @@ def _item_value_per_times_per_year(self, item, times_per_year=1):
         is_hourly = self.is_hourly and hasattr(item, "is_hourly") and item.is_hourly
 
         # If an item's value doesn't exist, use a value of 0
+        # TODO: is this behavior correct, or should it force gathering the value?
+        # What does a no-value item in the list represent?
         if hasattr(item, "value"):
             value = Decimal(item.value)
         else:
             value = Decimal(0)
 
-        # Use the appropriate cacluation
+        # Use the appropriate calculation
         if is_hourly:
             return (
                 value * Decimal(self.hours_per_period) * Decimal(frequency_to_use)
             ) / Decimal(times_per_year)
         else:
             return (value * Decimal(frequency_to_use)) / Decimal(times_per_year)
 
-    def total(self, times_per_year=1, source=None):
+    def total(
+        self, times_per_year: float = 1, source: Union[List[str], str] = None
+    ) -> Decimal:
         """
         Alias for ALItemizedJob.gross_total to integrate with ALIncomeList math.
         """
         return self.gross_total(times_per_year=times_per_year, source=source)
 
-    def gross_total(self, times_per_year=1, source=None):
+    def gross_total(
+        self, times_per_year: float = 1, source: Union[List[str], str] = None
+    ) -> Decimal:
         """
         Returns the sum of positive values (payments) for a given times_per_year.
         You can filter the items by `source`. `source` can be a string or a list.
         If you use sources from deductions, they will be ignored.
 
         Args:
-        kwarg: times_per_year {str | num} (Optional) Number of times per year you
+        kwarg: times_per_year {str | float} (Optional) Number of times per year you
```

Comment:
Sometimes this appears as `{float}` and sometimes as `{str | float}`. If that's intentional, it's fine with me.

Reply:
Sorry, I had fixed that in the final commit--I know it's tricky to view a diff spanning multiple commits in a PR